polyfill/lib/ecmascript.mjs

```diff
@@ -1914,19 +1914,11 @@ export const ES = ObjectAssign({}, ES2022, {
 
   ToTemporalTimeZoneSlotValue: (temporalTimeZoneLike) => {
     if (ES.Type(temporalTimeZoneLike) === 'Object') {
-      if (ES.IsTemporalTimeZone(temporalTimeZoneLike)) return temporalTimeZoneLike;
       if (ES.IsTemporalZonedDateTime(temporalTimeZoneLike)) return GetSlot(temporalTimeZoneLike, TIME_ZONE);
       if (ES.IsTemporalCalendar(temporalTimeZoneLike)) {
         throw new RangeError('Expected a time zone object but received a Temporal.Calendar');
       }
-      if (!('timeZone' in temporalTimeZoneLike)) return temporalTimeZoneLike;
-      temporalTimeZoneLike = temporalTimeZoneLike.timeZone;
-      if (ES.Type(temporalTimeZoneLike) === 'Object') {
-        if (ES.IsTemporalCalendar(temporalTimeZoneLike)) {
-          throw new RangeError('Expected a time zone object as the timeZone property but received a Temporal.Calendar');
-        }
-        if (!('timeZone' in temporalTimeZoneLike)) return temporalTimeZoneLike;
-      }
+      return temporalTimeZoneLike;
```

Comment:
What's the reason for this change?  Not saying it's bad, just wanted to understand the rationale.

Reply:
To summarize what we discussed today in the champions meeting, we had discussed changing what Calendar.from and TimeZone.from accept potentially as part of #1808. However, since it's separable from the main change (which is why I tacked it on in separate commits) we'll have that discussion separately as part of #2104. I'll remove the commits from this PR.